### PR TITLE
Oleksandr_Tsapin_HW8

### PR DESCRIPTION
**Pull request by OleksandrTsapin**

### CODE REVIEW

File: tsapin/HW8/task1.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    if operator == '*': return x * y
+    if operator == '/': return x / y
+    else: return 'Unknown operation'
+
+#check
+print(calculator(100,5,'+'))
+print(calculator(100,5,'-'))
+print(calculator(100,5,'*'))
+print(calculator(100,5,'/'))
+print(calculator(100,5,'//'))
```
Comment (katemalash) [inline tsapin/HW8/task1.py:18]: Good Decision

File: tsapin/HW8/task2.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# HW8_Task2
+# The user makes a deposit of n dollars for year at 10% per annum (each year the amount of his deposit 
+# increases by 10%. This money is added to the deposit amount, and next year there will also be interest on it).
+# Create a bank function that takes the arguments n and years and returns the amount that will be in the user's 
+# account.
+
+annual_interest = 0.1 #interest rate 10%
+
+def deposit(initial_balance, years):
+    return initial_balance * (1+annual_interest)**years
```
Comment (katemalash) [inline tsapin/HW8/task2.py:10]: nice optimization

File: tsapin/HW8/task2.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+     
+print(deposit(100,10))
```
Comment (katemalash) [inline tsapin/HW8/task2.py:12]: Perfect Decision

File: tsapin/HW8/task3.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# HW8_Task3
+# Create a date function that takes 3 arguments - day, month and year. Return True if such a date is in our 
+# calendar, and False - otherwise. 
+
+def date(day, month, year): #date format 'd.m.yyyy'
```
Comment (katemalash) [inline tsapin/HW8/task3.py:5]: Just advice: it would be better use `dd.mm.yyyy`

File: tsapin/HW8/task3.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        return False 
+
+while True:
+    print("--> ",
+          date(int(input("day: ")),
+               int(input("month: ")),
+               int(input("year: "))
+               )
+          )
+    print()
```
Comment (katemalash) [inline tsapin/HW8/task3.py:41]: Pretty good decision. You've done it without` date()` function. Nice.

File: tsapin/HW8/task4.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+# and False - otherwise. 
+
+def is_prime(number):
+    if number not in range(0,1001): print("Number shuold be in range from 0 to 1000")
+    elif number == 0: print("Not natural")
+    elif number == 1: return False
+    elif number == 2 or number == 3: return True
+    elif not number%2: return False
+    elif not number%3: return False
+    else: return True
```
Comment (katemalash) [inline tsapin/HW8/task4.py:12]: This algorithm is not correct. you check only possibility for dividing on 2 and three.

File: tsapin/HW8/task5.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    if year < first_leap_year: return False #NOT LEAP - less than 1584
+    elif year%4: return False #NOT LEAP - not multiple of 4
+    else:
+        if year%100: return True #LEAP - multiple of 4, not multiple of 100
+        elif not year%400: return True #LEAP - multiple of 4, 100 and 400
+        else: return False #NOT LEAP - multiple of 4, not multiple of 100 AND 400
+
+print("Is year leap?")
+while True:
+    print(is_year_leap(int(input("--> year: "))))
```
Comment (katemalash) [inline tsapin/HW8/task5.py:16]: Nice

**Review by katemalash — CHANGES_REQUESTED** [review]
Please correct the solution of task with the prime numbers.

File: tsapin/HW8/task3.py (hunk 4 above)
Comment (OleksandrTsapin) [reply tsapin/HW8/task3.py:5]: Sure. I checked - the app works with dd.mm.yyyy as well. So I'll just change the doc notation.

Comment (OleksandrTsapin) [reply tsapin/HW8/task3.py:5]: Sure. I checked - the app works with dd.mm.yyyy as well. So I'll just change the doc notation.

File: tsapin/HW8/task4.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+print("Is number prime?")
+
+while True:
+    print("\nEnter number from 0 to 1000:")
+    num = int(input("--> "))
+    
+    if num not in range(0,1001):
+        print("Number shuold be in range from 0 to 1000")
+    else:
+        print(is_prime(num))
```
Comment (katemalash) [inline tsapin/HW8/task4.py:33]: Pretty nice decision!

**Review by katemalash — APPROVED** [review]
Good Job. Now you can merge this pull-request.

**Pull request merged**